From ec2bab81b6b1cef4d844d872caaf3b83dcdf35c4 Mon Sep 17 00:00:00 2001
From: Maxim Uvarov <maxim.uvarov@linaro.org>
Date: Mon, 7 Sep 2020 15:32:18 +0100
Subject: [PATCH] efi/libstub: DRAM base calculation

qemu has secure ram memory region outside the system RAM. That memory
should not be taken under account when reserving areas for inird or
kernel.

Signed-off-by: Maxim Uvarov <maxim.uvarov@linaro.org>
---
 drivers/firmware/efi/libstub/efi-stub.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/firmware/efi/libstub/efi-stub.c b/drivers/firmware/efi/libstub/efi-stub.c
index 4aecafa325fd..191223a3e019 100644
--- a/drivers/firmware/efi/libstub/efi-stub.c
+++ b/drivers/firmware/efi/libstub/efi-stub.c
@@ -153,7 +153,8 @@ static unsigned long get_dram_base(void)
 	map.map_end = map.map + map_size;
 
 	for_each_efi_memory_desc_in_map(&map, md) {
-		if (md->attribute & EFI_MEMORY_WB) {
+		if (md->attribute & EFI_MEMORY_WB &&
+			md->type == EFI_CONVENTIONAL_MEMORY) {
 			if (membase > md->phys_addr)
 				membase = md->phys_addr;
 		}
-- 
2.17.1

